File: problems/problems_2972/solution.ts
function incremovableSubarrayCount(nums: number[]): number {
	const n: number = nums.length;
	let i: number = 0;
	while (i < n - 1 && nums[i] < nums[i + 1]) {
		i++;
	}
	if (i === n - 1) {
		return Math.floor(n * (n + 1) / 2);
	}
	let ans: number = i + 2;
	let j: number = n - 1;
	while (j == n - 1 || nums[j] < nums[j + 1]) {
		while (i >= 0 && nums[i] >= nums[j]) {
			i--;
		}
		ans += i + 2;
		j--;
	}
	return ans;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const nums: number[] = JSON.parse(inputValues[0]);
	return incremovableSubarrayCount(nums);
}

File: problems/problems_2972/solution.go
package problem2972

import (
	"encoding/json"
	"log"
	"strings"
)

func incremovableSubarrayCount(nums []int) int64 {
	n := len(nums)
	i := 0
	for i < n-1 && nums[i] < nums[i+1] {
		i++
	}
	if i == n-1 {
		return int64(n) * int64(n+1) / 2
	}
	ans := int64(i + 2)
	j := n - 1
	for j == n-1 || nums[j] < nums[j+1] {
		for i >= 0 && nums[i] >= nums[j] {
			i--
		}
		ans += int64(i + 2)
		j--
	}
	return ans
}

func Solve(inputJsonValues string) interface{} {
	inputValues := strings.Split(inputJsonValues, "\n")
	var nums []int

	if err := json.Unmarshal([]byte(inputValues[0]), &nums); err != nil {
		log.Fatal(err)
	}

	return incremovableSubarrayCount(nums)
}
